lib/compat/wordpress-6.6/post.php

```diff
@@ -33,7 +35,35 @@ function gutenberg_update_wp_template_labels( $labels ) {
  * @return object Object with all the labels as member variables.
  */
 function gutenberg_update_wp_template__part_labels( $labels ) {
+	$labels->add_new      = __( 'Add Template Part', 'gutenberg' );
+	$labels->add_new_item = __( 'Add Template Part', 'gutenberg' );
 	$labels->item_updated = __( 'Template part updated.', 'gutenberg' );
 	return $labels;
 }
 add_filter( 'post_type_labels_wp_template_part', 'gutenberg_update_wp_template__part_labels', 10, 1 );
+
+/**
+ * Updates the labels for the pattern post type.
+ *
+ * @param  object $labels Object with labels for the post type as member variables.
+ * @return object Object with all the labels as member variables.
+ */
+function gutenberg_update_wp_block_labels( $labels ) {
+	$labels->add_new      = __( 'Add Pattern', 'gutenberg' );
+	$labels->add_new_item = __( 'Add Pattern', 'gutenberg' );
+	return $labels;
+}
+add_filter( 'post_type_labels_wp_block', 'gutenberg_update_wp_block_labels', 10, 1 );
+
+/**
+ * Updates the labels for the page post type.
+ *
+ * @param  object $labels Object with labels for the post type as member variables.
+ * @return object Object with all the labels as member variables.
+ */
+function gutenberg_update_page_labels( $labels ) {
+	$labels->add_new      = __( 'Add Page', 'gutenberg' );
+	$labels->add_new_item = __( 'Add Page', 'gutenberg' );
+	return $labels;
+}
+add_filter( 'post_type_labels_page', 'gutenberg_update_page_labels', 10, 1 );
```

Comment:
I think ideally we should make this change directly in Core because it will impact more than just the site editor and will not be limited to these post types. For instance I don't see why we should change these labels for "pages" but not for "posts". So I'd leave these label changes out of this PR personally.

Reply:
>I can extract the labels update to remove the new for other post types too in a separate PR

I'd create a GB PR for that, but you suggest skip it altogether and only do it in core?

Reply:
The important thing is that it should update all the post types to stay consistent (not just the ones shown in the site editor). I suggested to do it in Core because it impacts the UI of WP-Admin entirely and Core might give more visibility to this change. It shouldn't be hidden in the "Add create pattern button" PR.

Reply:
For reference I opened a trac ticket about this last week; https://core.trac.wordpress.org/ticket/61219#ticket

Reply:
Extracted them and opened a core PR: https://github.com/WordPress/wordpress-develop/pull/6585.

Reply:
Great :)